dart/lib/src/exception_stacktrace_extractor.dart

```diff
@@ -0,0 +1,35 @@
+import 'protocol.dart';
+import 'sentry_options.dart';
+
+/// Sentry handles [Error.stackTrace] by default. For other cases
+/// extend this abstract class and return a custom [StackTrace] of your
+/// exceptions.
+///
+/// Implementing an extractor and providing it through
+/// [SentryOptions.addExceptionStackTraceExtractor] will enable the framework to
+/// extract the inner stacktrace and add it to [SentryException] when no other
+/// stacktrace was provided while capturing the event.
+///
+/// For an example on how to use the API refer to dio/DioStackTraceExtractor or the
+/// code below:
+///
+/// ```dart
+/// class ExceptionWithInner {
+///   ExceptionWithInner(this.innerException, this.innerStackTrace);
+///   Object innerException;
+///   StackTrace innerStackTrace;
+/// }
+///
+/// class ExceptionWithInnerStackTraceExtractor extends ExceptionStackTraceExtractor<ExceptionWithInner>  {
+///   @override
+///   StackTrace? cause(ExceptionWithInner error) {
+///     return error.innerStackTrace;
+///   }
+/// }
+///
+/// options.addExceptionStackTraceExtractor(ExceptionWithInnerStackTraceExtractor());
+/// ```
+abstract class ExceptionStackTraceExtractor<T> {
+  StackTrace? stackTrace(T error);
```

Comment:
```suggestion
  dynamic stackTrace(T error);
```

I don't like that too but a `StackTrace` can be in many forms.
Can be a `Chain`, can be a `Trace`, can be a `String` if coming from Platform exceptions, etc.

Reply:
Yea but `Chain`/`Trace` implement `StackTrace` and `String` can be converted with `StackTrace.fromString()`.
Native stacks are the only thing where I see the point :) Will change to keep it consistent with the rest of the SDK.

Reply:
Done

Reply:
yeah, the boilerplate from `StackTrace.fromString()` is suboptimal if the SDK can do it automatically anyway.